pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -131,7 +134,8 @@ func (p *PoolManager) UpdateMacAddressesForVirtualMachine(previousVirtualMachine
 	log.V(1).Info("UpdateMacAddressesForVirtualMachine: data",
 		"macmap", p.macPoolMap,
 		"podmap", p.podToMacPoolMap,
-		"currentMac", p.currentMac.String())
+		"currentMac", p.currentMac.String(),
+		"virtualMachine.Name", virtualMachine.Name)
 	if previousVirtualMachine == nil {
 		p.poolMutex.Unlock()
```

Comment:
I think this can be right next to Lock(),  either way Unlock() is called when this function returns

Reply:
since AllocateVirtualMachineMac() also uses Lock(), we want to avoid unnecessary double locking.